### PR TITLE
Decouple use_sequence_as_input_ids from has_hidden_states

### PR DESCRIPTION
**Pull request by wangyems**
### Description

A fix for parity issue in huggingface bart model with beam search
https://github.com/microsoft/onnxruntime/pull/12779

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.h [hunk 1]
```diff
@@ -54,8 +57,9 @@ class T5DecoderSubgraph : public Subgraph {
     return first_present_output_index_;
   }
 
-  int HasHiddenStates() const {
-    return has_hidden_state_;
+  bool UseSequenceAsInputIds() const {
+    // Custom model use sequence as input ids for decoder subgraph input.
+    return ParseEnvironmentVariableWithDefault<bool>(kUseSequenceAsInputIds, false);
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.h:62]: Environment variable is for testing only. Not recommend for production.

Shall we detect it from subgraph input_ids shape, like ['batch_size', 1] vs ['batch_size', 'sequence_length']?

Comment (wangyems) [reply onnxruntime/contrib_ops/cpu/transformers/subgraph_t5_decoder.h:62]: sounds good

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**